cmd/controller/main.go

```diff
@@ -211,14 +216,44 @@ func main() {
 		}
 	}
 
-	startUtilHttpServerOrDie(kubeClient, kubeconfigProvider, stopCh)
-	startControllerViaLeaderElectionOrDie(runOptions)
+	ctx, ctxDone := context.WithCancel(context.Background())
+	runOptions.ctx = ctx
+	var g run.Group
+	{
+		g.Add(func() error {
+			return startUtilHttpServer(kubeClient, kubeconfigProvider, ctx)
+		}, func(err error) {
+			ctxDone()
+		})
+	}
+	{
+		g.Add(func() error {
```

Comment:
I think we don't need this group. 😉

Reply:
this group listens for Ctrl-C, for example a user requested an app to be closed.

Reply:
If we will remove this group how can we handle such request ?